src/topp/MzTabExporter.cpp

```diff
@@ -437,38 +514,87 @@ namespace OpenMS
-//          MzTabString species = hit.getSpecies(); // Human readable name of the species
+         // protein_row.taxid = hit.getTaxonomyID(); // TODO add as meta value to protein hitNEWT taxonomy for the species.
+         // MzTabString species = hit.getSpecies(); // Human readable name of the species
             protein_row.database = db; // Name of the protein database.
             protein_row.database_version = db_version; // String Version of the protein database.
             protein_row.best_search_engine_score[1] = MzTabDouble(hit.getScore());
-//          MzTabParameterList search_engine; // Search engine(s) identifying the protein.
-//          std::map<Size, MzTabDouble>  best_search_engine_score; // best_search_engine_score[1-n]
-//          std::map<Size, std::map<Size, MzTabDouble> > search_engine_score_ms_run; // search_engine_score[index1]_ms_run[index2]
-//          MzTabInteger reliability;
-//          std::map<Size, MzTabInteger> num_psms_ms_run;
-//          std::map<Size, MzTabInteger> num_peptides_distinct_ms_run;
-//          std::map<Size, MzTabInteger> num_peptides_unique_ms_run;
-//          MzTabModificationList modifications; // Modifications identified in the protein.
-//          MzTabString uri; // Location of the protein’s source entry.
-//          MzTabStringList go_terms; // List of GO terms for the protein.
+         // MzTabParameterList search_engine; // Search engine(s) identifying the protein.
+         // std::map<Size, MzTabDouble>  best_search_engine_score; // best_search_engine_score[1-n]
+         // std::map<Size, std::map<Size, MzTabDouble> > search_engine_score_ms_run; // search_engine_score[index1]_ms_run[index2]
+         // MzTabInteger reliability;
+         // std::map<Size, MzTabInteger> num_psms_ms_run;
+         // std::map<Size, MzTabInteger> num_peptides_distinct_ms_run;
+         // std::map<Size, MzTabInteger> num_peptides_unique_ms_run;
+         // MzTabModificationList modifications; // Modifications identified in the protein.
+         // MzTabString uri; // Location of the protein’s source entry.
+         // MzTabStringList go_terms; // List of GO terms for the protein.
             double coverage = hit.getCoverage();
             protein_row.protein_coverage = coverage >= 0 ? MzTabDouble(coverage) : MzTabDouble(); // (0-1) Amount of protein sequence identified.
-//          std::vector<MzTabOptionalColumnEntry> opt_; // Optional Columns must start with “opt_”
-            // optional column
+         // std::vector<MzTabOptionalColumnEntry> opt_; // Optional Columns must start with “opt_”
+
+            // create opt_ columns for protein hit user values
+            for (set<String>::const_iterator mit = protein_hit_user_value_keys.begin(); mit != protein_hit_user_value_keys.end(); ++mit)
+            {
+              MzTabOptionalColumnEntry opt_entry;
+              const String& key = *mit;
+              opt_entry.first = String("opt_prt_") + key;
```

Comment:
What happens when metainfos on different levels (e.g. Feature and PeptideID) have the same name?